packages/core/data-transfer/lib/providers/local-strapi-destination-provider/restore.ts

```diff
@@ -0,0 +1,47 @@
+import type { ContentTypeSchema } from '@strapi/strapi';
+
+export type DeleteOptions = {
+  contentTypes?: ContentTypeSchema[];
+  uidsOfModelsToDelete?: string[];
+  conditions?: {
+    [contentTypeUid: string]: {
+      params: any;
+    };
+  };
+};
+
+export const deleteAllRecords = async (strapi: Strapi.Strapi, deleteOptions?: DeleteOptions) => {
+  const conditions = deleteOptions?.conditions ?? {};
+  const exceptions = [
+    'admin::permission',
+    'admin::user',
+    'admin::role',
+    'admin::api-token',
+    'admin::api-token-permission',
+  ];
+  const defaultUids = ['webhook', 'strapi::core-store'];
+  const uids = deleteOptions?.uidsOfModelsToDelete ?? defaultUids;
+  const allContentTypes: ContentTypeSchema[] = Object.values(strapi.contentTypes);
+  const defaultContentTypes: ContentTypeSchema[] = allContentTypes.filter(
+    (contentType) => !exceptions.includes(contentType.uid)
+  );
+  const contentTypes: ContentTypeSchema[] = deleteOptions?.contentTypes ?? defaultContentTypes;
+  let count = 0;
+
+  await Promise.all(
+    contentTypes.map(async (contentType) => {
+      const params = conditions[contentType.uid]?.params ?? {};
+      const result = await strapi?.entityService.deleteMany(contentType.uid, params);
+      count += result ? result.count : 0;
+    })
+  );
+
+  await Promise.all(
+    uids.map(async (uid) => {
+      const result = await strapi?.query(uid).deleteMany();
+      count += result.count;
+    })
+  );
+
+  return { count };
```

Comment:
If I understand correctly, uids are used to delete data at a low level while contentTypes are used with the entity service to remove all related data. Am I right?
If yes, maybe we could rename a bit the variables etc... It might be a bit confusing on a first read

Reply:
yes exactly, any suggestions for the namings ?
`modelsUids` maybe